src/proof/linked_list.rs

```diff
@@ -0,0 +1,262 @@
+//! Double LinkedList.
+//!
+//! This is not super fast implementation, but we just need something to be working for now.
+//!
+
+use std::cell::{Ref, RefCell, RefMut};
+use std::rc::Rc;
```

Comment:
Are there implementations for these in `core` lib? This won't work directly in runtime if we need this data structure.

Reply:
currently, we make no use of this LinkedList implementation, so I will remove it for now.